src/dataflow/src/source/kafka.rs

```diff
@@ -46,6 +47,92 @@ pub struct PreviousStats {
     consumer_lag: i64,
 }
 
+#[derive(Clone, Copy, Default)]
+pub struct BrokerRTTWindow {
+    min: i64,
+    max: i64,
+    avg: i64,
+    sum: i64,
+    cnt: i64,
+    stddev: i64,
+    p50: i64,
+    p75: i64,
+    p90: i64,
+    p95: i64,
+    p99: i64,
+    p99_99: i64,
+}
+
+impl BrokerRTTWindow {
+    /// Return the value required to negate the last value recorded for this window
+    fn negate(
+        self,
+        consumer_name: String,
+        source_id: SourceInstanceId,
+        broker_name: String,
+    ) -> MaterializedEvent {
+        MaterializedEvent::KafkaBrokerRTT {
+            consumer_name: consumer_name,
+            source_id: source_id,
+            broker_name: broker_name,
+            min: -self.min,
+            max: -self.max,
+            avg: -self.avg,
+            sum: -self.sum,
+            cnt: -self.cnt,
+            stddev: -self.stddev,
+            p50: -self.p50,
+            p75: -self.p75,
+            p90: -self.p90,
+            p95: -self.p95,
+            p99: -self.p99,
+            p99_99: -self.p99_99,
+        }
+    }
+    /// Update the value for window, returning a MaterializedEvent that represents the
+    /// difference between the previous values and the new values
+    fn update(
+        &mut self,
+        consumer_name: String,
+        source_id: SourceInstanceId,
+        broker_name: String,
+        stats: &Window,
+    ) -> MaterializedEvent {
+        let event = MaterializedEvent::KafkaBrokerRTT {
+            consumer_name: consumer_name,
+            source_id: source_id,
+            broker_name: broker_name,
+            min: stats.min - self.min,
```

Comment:
I'm surprised this isn't getting linted, but I'd expect this to be required to be:

```suggestion
        let event = MaterializedEvent::KafkaBrokerRTT {
            consumer_name,
            source_id,
            broker_name,
            min: stats.min - self.min,
```

Since it's not getting linted, it's fine.

Reply:
We removed a bunch of Clippy style nits recently because people (including me) hated them. The only lints that should still be turned on are ones that actually find bugs or perf issues.

Reply:
Presence of the Clippy Hint aside, I like your suggestion @quodlibetor!